mlir/lib/Dialect/MIOpen/Generator/IsaNameParser.cpp

```diff
@@ -0,0 +1,128 @@
+//===- IsaNameParser.cpp - MLIR to C++ option parsing ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements isa name string parser
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/MIOpen/Generator/IsaNameParser.h"
+#include "llvm/Support/Error.h"
+
+#include <cstring>
+#include <numeric>
+#include <vector>
```

Comment:
can we not depend on `vector`, but on `llvm::SmallVector`?

Reply:
Sure

Reply:
Done in 7704bf2